Lib/fractions.py

```diff
@@ -429,14 +430,22 @@ def _div(a, b):
 
     def _floordiv(a, b):
         """a // b"""
-        return math.floor(a / b)
+        return (a.numerator * b.denominator) // (a.denominator * b.numerator)
 
     __floordiv__, __rfloordiv__ = _operator_fallbacks(_floordiv, operator.floordiv)
 
+    def _divmod(a, b):
+        """(a // b, a % b)"""
+        da, db = a.denominator, b.denominator
+        div, n_mod = divmod(a.numerator * db, da * b.numerator)
+        return div, Fraction(n_mod, da * db)
+
+    __divmod__, __rdivmod__ = _operator_fallbacks(_divmod, divmod)
+
     def _mod(a, b):
         """a % b"""
-        div = a // b
-        return a - b * div
+        da, db = a.denominator, b.denominator
+        return Fraction((a.numerator * db) % (b.numerator * da), da * db)
```

Comment:
Why do you create the da and db? why don't use directly b.denominator and a.denominator? :)

Reply:
For clarity and for performance. Using a local variable is faster than looking up an attribute.

Reply:
Great. So, this way, why don't create an a.numerator and b.numerator local variable?

Reply:
I guess because `a.numerator` and `b.numerator` are used only once